engines/director/lingo/xlibs/msfile.cpp

```diff
@@ -0,0 +1,335 @@
+	{"getFileList",         MSFile::m_getFileList,         1, 1, 400},
+	{"getDirList",          MSFile::m_getDirList,          1, 1, 400},
+	{"getDiskFree",         MSFile::m_getDiskFree,         1, 1, 400},
+	{"getVolLabel",         MSFile::m_getVolLabel,         1, 1, 400},
+	{"createDir",           MSFile::m_createDir,           1, 1, 400},
+	{"removeDir",           MSFile::m_removeDir,           1, 1, 400},
+	{"getCurrDir",          MSFile::m_getCurrDir,          0, 1, 400},
+	{"setCurrDir",          MSFile::m_setCurrDir,          1, 1, 400},
+	{"removeFile",          MSFile::m_removeFile,          1, 1, 400},
+	{"renameFile",          MSFile::m_renameFile,          2, 1, 400},
+	{"copyFile",            MSFile::m_copyFile,            2, 1, 400},
+	{"copyFiles",           MSFile::m_copyFiles,           3, 1, 400},
+	{"setAppINI",           MSFile::m_setAppINI,           4, 1, 400},
+	{"getAppINI",           MSFile::m_getAppINI,           3, 1, 400},
+	{"getWindowsDirectory", MSFile::m_getWindowsDirectory, 0, 1, 400},
+	{"getWinVer",           MSFile::m_getWinVer,           0, 1, 400},
+	{"restartWindows",      MSFile::m_restartWindows,      0 ,1, 400},
+	{"openFileDlg",         MSFile::m_openFileDlg,         3, 1, 400},
+	{"saveFileDlg",         MSFile::m_saveFileDlg,         3, 1, 400},
+	{"getOFileName",        MSFile::m_getOFileName,        3, 1, 400},
+	{"getSFileName",        MSFile::m_getSFileName,        3, 1, 400},
+	{"displayDlg",          MSFile::m_displayDlg,          3, 1, 400},
+	{"progMgrAdd",          MSFile::m_progMgrAdd,          4, 1, 400},
+	{nullptr, nullptr, 0, 0, 0}};
+
+static const BuiltinProto xlibBuiltins[] = {
+	{nullptr, nullptr, 0, 0, 0, VOIDSYM}};
+
+MSFileObject::MSFileObject(ObjectType objType) : Object<MSFileObject>("MSFile") {
+	_objType = objType;
+	_lastError = msErrorNone;
+}
+
+MSFileObject::MSFileObject(const MSFileObject &obj) : Object<MSFileObject>(obj) {
+	_objType = obj.getObjType();
+	_lastError = msErrorNone;
+}
+
+MSFileObject::~MSFileObject() {
+}
+
+void MSFileObject::dispose() {
+	_disposed = true;
+}
+
+bool MSFileObject::hasProp(const Common::String &propName) {
+	return (propName == "name");
+}
+
+Datum MSFileObject::getProp(const Common::String &propName) {
+	if (propName == "name")
+		return Datum(MSFile::xlibName);
+	warning("MSFile::getProp: unknown property '%s'", propName.c_str());
+	return Datum();
+}
+
+void MSFile::m_new(int nargs) {
+	Datum result = g_lingo->_state->me;
+	g_lingo->push(result);
+}
+
+XOBJSTUB(MSFile::m_dispose, 0)
+XOBJSTUB(MSFile::m_name, "")
+XOBJSTUB(MSFile::m_getFullDirList, "")
+XOBJSTUB(MSFile::m_getFileList, "")
+XOBJSTUB(MSFile::m_getDirList, "")
+
+void MSFile::m_getDiskFree(int nargs) {
+	// We don't care about the drive letter. We will always just return 500 MB.
+	g_lingo->pop();
+
+	// 500 MB in KB
+	g_lingo->push(Datum(524288000));
+}
+
+void MSFile::m_getVolLabel(int nargs) {
+	Datum d = g_lingo->pop();
+
+	Common::String volLabel = "";
```

Comment:
```suggestion
	Common::String volLabel;
```
It is already initialized with a zero-length string.

Reply:
Ah ok I can make that change, thanks for pointing that out.